### DISCUSSION TITLE
Intergrated Foot Controller

### DISCUSSION THREAD

**Discussion by markus625** [Builty/TonexOneController#14] [Ideas]
Would be great if you could combine this project with a DIY midi foot controller so it could be in one unit, actually the switches wouldnt need to send midi messages just the messages the tonex one undersands over the USB

**Comment by markus625** [comment]
Actually you could implement a ESP32-S3 1.91inch AMOLED Display Development Board.

https://www.waveshare.com/esp32-s3-amoled-1.91.htm

https://www.waveshare.com/img/devkit/ESP32-S3-Touch-AMOLED-1.91/ESP32-S3-Touch-AMOLED-1.91-details-3-3.jpg

And combine this with some foot switches akin to the project below .

https://github.com/alf45tar/PedalinoMini

**Comment by Builty** [comment]
I actually have a simple 2 footswitch pedal running on my pedal board. Just an off the shelf one with 6.5mm jack, connected to the Waveshare digital inputs. This is in the standard release for the display version.

That amoled board doesn't apper to be suitable. To support USB OTG it needs to provide 5v power out of the USB port. That one looks to have a diode that prevents that.
The ones I've chosen don't have this issue.

There are certainly plenty of options here though. Pedalino might be able to have this code integrated into it directly, if the OTG part is OK

**Reply by markus625** [reply]
Ah I see thanks for the explanation and reply .

**Comment by Builty** [comment]
Code has been changed now to support 4 footswitches directly wired. Uses bank selection like the MVave Chocolate.

I made a small box with 4 switches just for testing, but with footswitches in a stronger box it could remove the need for a bluetooth pedal.
![image](https://github.com/user-attachments/assets/a1686f1c-3655-4efc-bbf9-9e619675579a)

**Reply by jesusglezalv** [reply]
Is this only available for the Dev-kit C?

**Reply by Builty** [reply]
4 buttons is supported on all platforms except for the Waveshare 4.3B. It doesn't have enough spare inputs. 
All of the others have enough and can use the 4 buttons.
